### PR TITLE
PLT-9168 : Remove the wallet/Nodes/SingleAddressWallet package

### PR DESCRIPTION
**Pull request by nhenin**
## Summary by CodeRabbit

- **New Features**
    - Enhanced health check endpoint to provide detailed runtime status.
    - Expanded testing capabilities with new Jest configurations for end-to-end tests.
    - Added new HTTP request functions with improved error handling.
    - Introduced management and manipulation of Lucid assets.
    - New functionality for retrieving runtime status and health information.
    - Added comprehensive test environment setup for end-to-end testing.
    - Provided a utility to generate new seed phrases for wallets.

- **Improvements**
    - Reorganized and improved development documentation.
    - Streamlined runtime lifecycle functions for better usability.
    - Updated export names for clarity and consistency.
    - Improved logging utilities for Wallet Test API.

- **Bug Fixes**
    - Fixed issues with environment configuration in test setups.
    - Corrected test cases to align with updated functionality.

- **Documentation**
    - Expanded the How To Develop guide with additional instructions.
    - Updated README for the testing-kit package.

- **Refactor**
    - Refined the control flow and logic in various test specifications.
    - Consolidated and optimized import statements across packages.

- **Chores**
    - Updated npm ignore settings to exclude unnecessary directories.
    - Renamed debug configurations for clarity.

- **Style**
    - Adjusted styles for consistency across development tools.

- **Tests**
    - Configured Jest for both unit and end-to-end testing.
    - Enhanced test cases to reflect new features and improvements.

- **Revert**
    - No changes to report.

### CODE REVIEW

File: packages/testing-kit/src/wallet/lucid/provisionning.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+              aDistribution[2],
+              toAssetsToTransfer(aDistribution[3])
+            )
+            .payToAddress(aDistribution[2], { lovelace: 5_000_000n })
+            .payToAddress(aDistribution[2], { lovelace: 5_000_000n })
+            .payToAddress(aDistribution[2], { lovelace: 5_000_000n }) // add a Collateral
+      )
+    );
+
+    var result: ProvisionResponse = {};
```
Comment (bjornkihlberg) [inline packages/testing-kit/src/wallet/lucid/provisionning.ts:121]: Is there a particular reason you're using `var` here? If you need to mutate you should probably use `let`. `result` isn't actually mutated directly, (think of it as a pointer in this case) so you could use `const`.

File: packages/testing-kit/src/wallet/lucid/provisionning.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    distributions.map(([participant, wallet, , assetsProvisionned]) => {
+      result[participant] = {
+        wallet: wallet,
+        assetsProvisionned: assetsProvisionned,
+      };
+    });
```
Comment (bjornkihlberg) [inline packages/testing-kit/src/wallet/lucid/provisionning.ts:127]: We don't need to mutate at all actually
```suggestion
    const result = Object.fromEntries(
      distributions.map(([participant, wallet, , assetsProvisionned]) =>
        [participant, { wallet, assetsProvisionned }]));
```

Also `assetsProvisionned` should be `assetsProvisioned`.

File: packages/wallet/test/wallet.spec.ts [hunk 3]
```diff
@@ -1,7 +1,6 @@
-import { getPrivateKeyFromHexString } from "@marlowe.io/wallet/nodejs";
 
 describe("wallet", () => {
   it("succeeds", () => {
-    expect(getPrivateKeyFromHexString).toBeDefined();
+    expect(true).toBeDefined();
```
Comment (bjornkihlberg) [inline packages/wallet/test/wallet.spec.ts:4]: This is redundant, it will always succeed.

**Review by bjornkihlberg — CHANGES_REQUESTED** [review]
Some things could be cleaned up and written with a more functional programming style.

File: packages/runtime/client/rest/src/runtime/version.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+export const runtimeVersion = (s: string) =>
+  unsafeEither(RuntimeVersionGuard.decode(s));
+
+export type CompatibleRuntimeVersion = "0.0.6" | "0.0.5";
+
+export const CompatibleRuntimeVersionGuard: t.Type<
+  CompatibleRuntimeVersion,
+  string
+> = t.union([t.literal("0.0.6"), t.literal("0.0.5")]);
```
[the remark this answers is not in the document]
Comment (hrajchert) [reply packages/runtime/client/rest/src/runtime/version.ts:24]: apparently

File: doc/howToDevelop.md [hunk 5]
```diff
@@ -1,53 +1,153 @@
...
+```
+2. Copy/paste the words within quotes in the env file.
 
-In order to run the E2E tests you need to create a `./env/.env.test` file that points to a working version of the Marlowe runtime and a working Blockfrost instance and a faucet PK.
+#### How to add tAda to the Bank Wallet via a faucet ? 
 
-If you haven't done it before, go to https://blockfrost.io/ and create a free-tier account. Then, create a project and copy the project ID. Blockfrost is a Lucid dependency, eventually when
-we migrate to a different library this wont be necessary.
+1. Retrieve your Bank Wallet payment address
+2. Go to https://docs.cardano.org/cardano-testnet/tools/faucet ask for test Ada on this address.
```
Comment (hrajchert) [inline doc/howToDevelop.md:131]: The genSeedPhrase does not print the address, so this task is a little bit cumbersome. Please print the address in genSeedPhrase

**Pull request merged**